src/data/set/basic.lean

```diff
@@ -1,25 +1,73 @@
 /-
 Copyright (c) 2014 Jeremy Avigad. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Author: Jeremy Avigad, Leonardo de Moura
+Authors: Jeremy Avigad, Leonardo de Moura
 -/
+
 import tactic.basic tactic.finish data.subtype logic.unique
-open function
 
-/-! # Basic properties of sets
+/-!
+
+# Basic properties of sets
+
+A set in Lean is a subset of a type. This file provides some basic definitions related to sets
```

Comment:
I think that first sentence is misleading (and meaningless as it stands). I know what you mean, but shouldn't you try something like: "Sets in mathlib are homogeneous, all their elements have the same type. Sets whose elements have type `X` are thus defined as `set X := X \to Prop`. Of course this definition is in the core library, but we have no better place to comment on it. You can also add a link to the ZFC encoding in mathlib, for people who want to do set theory instead of using sets.

Reply:
Thanks.

Reply:
Perhaps we could even add, after "set X := ...", something like "That is, sets are represented by their membership function. Note that this function need not be decidable."